scripts/fmt.sh

```
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+run_format_checks() {
+    local target_log=$1
+    local cargo_opts=$2
+
+    log "Running code formatting check for $target_log..."
+    cargo fmt --all -- --check
+
+    log "Running static analysis (clippy) for $target_log..."
+    cargo clippy $cargo_opts --all-targets --all-features -- -D warnings
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)